api/v1/verticadb_webhook.go

```diff
@@ -1183,6 +1189,52 @@ func (v *VerticaDB) validateSandboxes(allErrs field.ErrorList) field.ErrorList {
 	return v.validateSubclustersInSandboxes(allErrs)
 }
 
+// validateProxyconfig validates if provided proxy info is correct
+func (v *VerticaDB) validateProxyconfig(allErrs field.ErrorList) field.ErrorList {
+	// check if we are using proxy deployments
+	if vmeta.UseVProxy(v.Annotations) {
+		// check if proxy set and image must be non-empty
+		if v.Spec.Proxy == nil {
+			err := field.Invalid(field.NewPath("spec").Child("proxy"),
+				"",
+				"proxy is not set")
+			allErrs = append(allErrs, err)
+		}
+		if v.Spec.Proxy != nil && v.Spec.Proxy.Image == "" {
+			err := field.Invalid(field.NewPath("spec").Child("proxy").Child("image"),
+				v.Spec.Proxy.Image,
+				"proxy.image cannot be empty")
+			allErrs = append(allErrs, err)
+		}
+
+		// check if replica is set and value is valid
+		for i := range v.Spec.Subclusters {
+			sc := &v.Spec.Subclusters[i]
+			// proxy replicas must be >= 0(at all times)
+			if *sc.Proxy.Replicas < 0 {
+				err := field.Invalid(
+					field.NewPath("spec").Child("subclusters").Index(i).Child("proxy").Child("replicas"),
+					sc.Proxy.Replicas,
+					fmt.Sprintf("subcluster %q has an invalid value %d for the proxy replica",
+						sc.Name, *sc.Proxy.Replicas))
+				allErrs = append(allErrs, err)
+			}
+		}
+
+		// check if log level annotaion value valid
+		proxyLogLevel := vmeta.GetVProxyLogLevel(v.Annotations)
+		if !slices.Contains(validProxyLogLevel, proxyLogLevel) {
+			prefix := field.NewPath("metadata").Child("annotations")
+			errMsg := fmt.Sprintf("annotation %s value is not valid", vmeta.VProxyLogLevelAnnotation)
```

Comment:
`"annotation %s value is not valid, please use one of the following values %v, validProxyLogLevel"`

Reply:
Updated in https://github.com/vertica/vertica-kubernetes/pull/1000/commits/42ee9713632436e3a9b76048fe6e12035d6bc83d